src/Adapter/Smarty/SmartyCacheConfiguration.php

```diff
@@ -70,8 +70,8 @@ public function updateConfiguration(array $configuration)
     {
         if ($this->validateConfiguration($configuration)) {
             $this->configuration->set('PS_SMARTY_FORCE_COMPILE', $configuration['template_compilation']);
-            $this->configuration->set('PS_SMARTY_CACHE', $configuration['cache']);
-            $this->configuration->set('PS_SMARTY_LOCAL', $configuration['multi_front_optimization']);
+            $this->configuration->set('PS_SMARTY_CACHE', (bool)$configuration['cache']);
+            $this->configuration->set('PS_SMARTY_LOCAL', (bool)$configuration['multi_front_optimization']);
```

Comment:
same here, I'm not easy with new PHP code fixing possible database issues! How about a $configuration->getBoolean('KEY') instead?

Reply:
Ah, realy nice suggestion here!

Reply:
We could reuse the same features as the ParameterBag from Symfony, but as an interface with the database. https://api.symfony.com/3.0/Symfony/Component/HttpFoundation/ParameterBag.html

Reply:
agree :)